src/libs/api.ts

```diff
@@ -1,12 +1,23 @@
-import axios from 'axios';
+import axios, { AxiosInstance } from 'axios';
+
+let apiInstance: AxiosInstance | null;
 
 export const api = () => {
-  const api = axios.create({
-    baseURL: import.meta.env.VITE_APP_BASE_URL,
-  });
+  if (!apiInstance) {
+    apiInstance = axios.create({
+      baseURL: import.meta.env.VITE_APP_BASE_URL,
+    });
+  }
 
   const token = sessionStorage.getItem('token');
-  api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+  if (token) {
+    const headerToken =
+      apiInstance.defaults.headers.common.Authorization?.toString();
```

Comment:
ASK: 여기 타입을 nullable로 설정해준 이유가 뭘까용!?

Reply:
아루밍이 말해준 그 이유가 맞긴 합니다 ..! 값이 null이거나 undefined일 수 있는데 AxiosHeaderValue를 string으로 바꿔주려고 해서 옵셔널 체이닝이 필요했었네요.. 그래서 생각해본게 headerToken의 타입을 string이 아닌 AxiosHeaderValue로 두고 split이전에 string으로 바꿔주면 옵셔널 체이닝 붙이지 않아도 되어서 요렇게 바꿔보았습니다!

```suggestion
      const headerToken = apiInstance.defaults.headers.common.Authorization;

    if (!headerToken || token !== headerToken.toString().split(' ')[1]) {
      apiInstance.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    }
  }
```

Reply:
오!! 저두 이게 훨씬 좋은 것 같아요 !! 뭔가 각자 역할이 더 명확히 분리된 느낌 ?! 역시 천재 개발자 도리네여 ㄷㄷ